presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionTreeUtils.java

```diff
@@ -112,4 +123,49 @@ public static boolean isEqualComparisonExpression(Expression expression)
     {
         return expression instanceof ComparisonExpression && ((ComparisonExpression) expression).getOperator() == ComparisonExpression.Operator.EQUAL;
     }
+
+    static Optional<EnumType> tryResolveEnumLiteralType(QualifiedName qualifiedName, TypeManager typeManager)
+    {
+        Optional<QualifiedName> prefix = qualifiedName.getPrefix();
+        if (!prefix.isPresent()) {
+            // an enum literal should be of the form `MyEnum.my_key`
+            return Optional.empty();
+        }
+        try {
+            Type baseType = typeManager.getType(parseTypeSignature(prefix.get().toString()));
+            if (baseType instanceof EnumType) {
+                return Optional.of((EnumType) baseType);
+            }
+        }
+        catch (IllegalArgumentException e) {
+            return Optional.empty();
+        }
+        return Optional.empty();
+    }
+
+    private static boolean isEnumLiteral(DereferenceExpression node, Type nodeType)
+    {
+        if (!(nodeType instanceof EnumType)) {
+            return false;
+        }
+        QualifiedName qualifiedName = DereferenceExpression.getQualifiedName(node);
+        requireNonNull(qualifiedName, "qualifiedName is null");
+        Optional<QualifiedName> prefix = qualifiedName.getPrefix();
+        return prefix.isPresent()
+                && prefix.get().toString().equalsIgnoreCase(nodeType.getTypeSignature().getBase());
+    }
+
+    public static Optional<Object> tryResolveEnumLiteral(DereferenceExpression node, Type nodeType)
+    {
+        QualifiedName qualifiedName = DereferenceExpression.getQualifiedName(node);
+        checkArgument(qualifiedName != null, "qualifiedName is null");
```

Comment:
You can remove this. the `isEnumLiteral` function already checks this. Or move it after the `if` statement.